File: java/Sphere.java
package my.languages;

public class Sphere extends Circle {
  private double radius;
  
  public Sphere() {
    radius = 0;
  }
  
  public void setRadius(double r) {
    radius = r;
  }
  
  public double getRadius() {
    return radius;
  }
  
  public double getDiameter() {
      return (2.0 * radius);
  }

  public double getCircumference() {
      return (2.0 * Math.PI * radius);
  }

  public double getArea() {
      return (4.0 * Math.PI * Math.pow(radius, 2.0));
  }
  
  public double getVolume() {
    return (4.0/3.0 * Math.PI * Math.pow(radius, 3.0));
  }
}

File: python/sphere.py
import math
from circle import Circle

class Sphere(Circle):
    def __init__(self):
        super().__init__()

    def set_radius(self, radius):
        super().set_radius(radius)

    def get_radius(self):
        return super().get_radius()

    def get_diameter(self):
        return super().get_diameter()

    def get_circumference(self):
        return super().get_circumference()

    def get_area(self):
        return 4 * super().get_area()

    def get_volume(self):
        return 4/3 * math.pi * math.pow(self.radius, 3)

